src/MentionsTextInput.js

```diff
@@ -262,9 +262,10 @@ export default class MentionsTextInput extends Component {
       return;
     }
 
+    const nameRegex = this.props.nameRegex ? this.props.nameRegex : `[a-zA-Z0-9_-]*`;
     const start = this.triggerMatrix[index][0];
     const end = this.triggerMatrix[index][1];
-    const pattern = new RegExp(`${this.props.trigger}[a-zA-Z0-9_-]*`, `gi`);
+    const pattern = new RegExp(`${this.props.trigger}${nameRegex}`, `gi`);
```

Comment:
again I would avoid back tics (in ``gi``whenever string interpolation in not taking place. Otherwise someone might replace the string without replacing the back ticks end up with inadvertent string interpolation if their string includes a dollar sign, for example.